route-rs-afpacket/tests/integration_test.rs

```diff
@@ -0,0 +1,174 @@
+#![cfg(target_os = "linux")]
+
+use afpacket;
+use rand::{self, Rng};
+use route_rs_afpacket::{AfPacketInputLink, AfPacketOutputLink};
+use route_rs_packets as packets;
+use route_rs_runtime::{
+    link::LinkBuilder,
+    utils::test::{
+        harness::{initialize_runtime, test_link},
+        packet_generators::immediate_stream,
+    },
+};
+use std::{ffi::CString, net, time::Duration};
+use tokio::{self, time};
+
+fn create_socket() -> (afpacket::SendHalf, afpacket::RecvHalf) {
+    let iface = {
+        let iface = CString::new("lo");
+        assert!(iface.is_ok());
+        iface.unwrap()
+    };
+    let sock = {
+        let sock = afpacket::AsyncBoundSocket::from_interface(&iface);
+        assert!(sock.is_ok());
+        sock.unwrap()
+    };
+
+    sock.split()
+}
+
+fn make_packet(rng: &mut impl Rng) -> Vec<u8> {
+    let body = {
+        let mut body = vec![0; 64];
+        rng.fill(&mut body[..]);
+        body
+    };
+    let mut udp_pkt = packets::UdpSegment::empty();
+    udp_pkt.set_src_port(3001);
+    udp_pkt.set_dest_port(3002);
+    udp_pkt.set_payload(&body);
+    let mut ipv4_pkt = packets::Ipv4Packet::encap_udp(udp_pkt);
+    ipv4_pkt.set_src_addr(net::Ipv4Addr::new(10, 0, 0, 1));
+    ipv4_pkt.set_dest_addr(net::Ipv4Addr::new(10, 0, 0, 2));
+    ipv4_pkt.set_ttl(2);
+    let mut eth_pkt = packets::EthernetFrame::encap_ipv4(ipv4_pkt);
+    eth_pkt.set_src_mac(packets::MacAddr::new([0xac, 0x67, 0x3d, 0xfa, 0xbc, 0xdd]));
+    eth_pkt.set_dest_mac(packets::MacAddr::new([0xac, 0x67, 0x3d, 0xfa, 0xbc, 0xdf]));
+
+    eth_pkt.data
+}
+
+#[test]
+#[should_panic]
+#[ignore]
```

Comment:
Leave a comment why this test can't be run by default, maybe with steps to run the test by hand.

Reply:
All the other ignored tests should follow the same pattern